runhouse/main.py

```diff
@@ -590,159 +592,149 @@ def status(
-    not_running_clusters = sorted(
-        not_running_clusters,
-        key=lambda x: (x["Last Active (UTC)"] is None, x["Last Active (UTC)"]),
-        reverse=True,
-    )
+    # not_running_clusters = sorted(
+    #     not_running_clusters, key=lambda x: x["Last Active (UTC)"], reverse=True
+    # )
     running_clusters = sorted(
-        running_clusters,
-        key=lambda x: (x["Last Active (UTC)"] is None, x["Last Active (UTC)"]),
-        reverse=True,
+        running_clusters, key=lambda x: x["Last Active (UTC)"], reverse=True
     )
 
-    # creating the clusters table
-    total_clusters = len(clusters_in_den)
-    table_title = f"[bold cyan]{rns_client.username}'s Clusters (Running: {len(running_clusters)}, Total: {total_clusters})[/bold cyan]"
-    table = Table(title=table_title)
+    filters_requested: bool = show_all or since or cluster_status
 
-    # Add columns to the table
-    table.add_column("Name", justify="left", no_wrap=True)
-    table.add_column("Cluster Type", justify="center", no_wrap=True)
-    table.add_column("Status", justify="left")
-    table.add_column("Last Active (UTC)", justify="left")
+    clusters_to_print = all_clusters if filters_requested else running_clusters
 
-    # TODO: will be used if we'll need to print not-running clusters
-    # all_clusters = running_clusters + not_running_clusters
+    if show_all:
+        # if user requesting all den cluster, limit print only to 50 clusters max.
+        clusters_to_print = clusters_to_print[: (min(len(clusters_to_print), 50))]
 
-    for rh_cluster in running_clusters:
-        last_active_at = rh_cluster.get("Last Active (UTC)")
+    # creating the clusters table
+    table = _create_output_table(
+        total_clusters=len(all_clusters),
+        running_clusters=len(running_clusters),
+        displayed_clusters=len(clusters_to_print),
+        filters_requested=filters_requested,
+    )
 
-        # Print Running clusters that were active it the last 24 hours
-        now = datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc)
-        if (now - last_active_at).total_seconds() <= LAST_ACTIVE_AT_TIMEFRAME:
-            table.add_row(
-                rh_cluster.get("Name"),
-                rh_cluster.get("Cluster Type"),
-                StatusColors.get_status_color(rh_cluster.get("Status")),
-                str(last_active_at).split("+")[
-                    0
-                ],  # The split is required to remove the offset (according to UTC)
-            )
+    _add_clusters_to_output_table(
+        table=table, clusters=clusters_to_print, filters_requested=filters_requested
+    )
 
     console.print(table)
 
-    if len(sky_clusters) > 0:
-        console.print(
-            f"There are {len(sky_clusters)} live cluster(s) that are not saved in Den. For more information, please run [bold italic]sky status -r[/bold italic]."
-        )
+    _print_msg_live_sky_clusters(sky_clusters=len(sky_clusters))
```

Comment:
small nit - if only using in one place no need for a separate helper for this print?